api/package.json

```diff
@@ -0,0 +1,22 @@
+{
+    "name": "@microsoft/vscode-azureresources-api",
```

Comment:
`@microsoft/vscode-azureresources-api` is my favorite name because to me it makes the most sense that the API for the Azure Resources extension would be listed as `vscode-azureresources-api`. Only unfortunate part is that the extension id is actually `vscode-azureresourcegroups`, but I think it's fine to base the API package on the extension name and not the id.

Also `@microsoft/vscode-azureresourcegroups-api` is really long.

My second favorite is `@microsoft/vscode-azext-host-api` because it's shorter, and also doesn't have to deal with the naming/id awkwardness.